src/Symfony/Component/Translation/Bridge/Crowdin/CrowdinProvider.php

```diff
@@ -0,0 +1,425 @@
+
+            if (!$fileId) {
+                continue;
+            }
+
+            foreach ($locales as $locale) {
+                if ($locale !== $this->defaultLocale) {
+                    $content = $this->exportProjectTranslations($locale, $fileId);
+                } else {
+                    $content = $this->downloadSourceFile($fileId);
+                }
+
+                $translatorBag->addCatalogue($this->loader->load($content, $locale, $domain));
+            }
+        }
+
+        return $translatorBag;
+    }
+
+    public function delete(TranslatorBagInterface $translatorBag): void
+    {
+        $filesMap = $this->getFilesMap();
+
+        foreach ($translatorBag->getCatalogues() as $catalogue) {
+            if ($catalogue->getLocale() !== $this->defaultLocale) {
+                continue;
+            }
+
+            foreach ($catalogue->getDomains() as $domain) {
+                $fileId = $this->getFileIdByDomain($filesMap, $domain);
+
+                if (!$fileId) {
+                    continue;
+                }
+
+                $stringsMap = $this->mapStrings($fileId);
+
+                foreach ($catalogue->all() as $messages) {
+                    foreach ($messages as $id => $message) {
+                        if (!isset($stringsMap[$id])) {
+                            continue;
+                        }
+
+                        $this->deleteString($stringsMap[$id]);
+                    }
+                }
+            }
+        }
+    }
+
+    private function getFilesMap(): array
+    {
+        $result = [];
+
+        $fileList = $this->getFileList();
+
+        foreach ($fileList as $file) {
+            $result[$file['data']['name']] = $file['data']['id'];
+        }
+
+        return $result;
+    }
+
+    private function getFileIdByDomain(array $filesMap, string $domain): ?int
+    {
+        return $filesMap[sprintf('%s.%s', $domain, 'xlf')] ?? null;
+    }
+
+    private function mapStrings(int $fileId): array
+    {
+        $result = [];
+
+        $limit = 500;
+        $offset = 0;
+
+        do {
+            $strings = $this->listStrings($fileId, $limit, $offset);
+
+            foreach ($strings as $string) {
+                $result[$string['data']['text']] = $string['data']['id'];
+            }
+
+            $offset += $limit;
+        } while (\count($strings) > 0);
+
+        return $result;
+    }
+
+    private function addFile(string $domain, string $content): array
+    {
+        $storageId = $this->addStorage($domain, $content);
+
+        /**
+         * @see https://support.crowdin.com/api/v2/#operation/api.projects.files.getMany (Crowdin API)
+         * @see https://support.crowdin.com/enterprise/api/#operation/api.projects.files.getMany (Crowdin Enterprise API)
+         */
+        $response = $this->client->request('POST', sprintf('https://%s/projects/%s/files', $this->endpoint, $this->projectId), [
+            'headers' => [
```

Comment:
If you use `json` instead of `body` here, you don't need to manually add the header, HttpClient will do it for you. (same for all similar requests)

Reply:
Tried this and Crowdin API returned the `415 Unsupported Media Type` response. Without `Content-Type` header and with `body` passed as array request params was interpreted as a query string, not JSON.

Reply:
Did you tried something like this? https://github.com/symfony/symfony/pull/40927/files#diff-c2f25a171f29f2922f7bc4c3716f02ad315234a9afcab835de4b6c1896c150ccR146-R154

Reply:
Oh, I didn't notice that the options array should contain the `json` key. I tried it using the `body` key.